include/parlay/alloc.h

```diff
@@ -154,7 +154,7 @@ inline void p_free(void* ptr) {
 //    std::vector<int, parlay::allocator<int>>
 //
 template <typename T>
-struct allocator {
+struct PARLAY_TRIVIALLY_RELOCATABLE allocator {
```

Comment:
This annotation isn't needed, AFAICT. `allocator` is a Rule-of-Zero type with no data members. (Its templated constructor doesn't matter because it still has an implicitly defaulted copy ctor and move ctor.)